src/DataLib/fendl32B_retrofit/tendl_processing.py

```diff
@@ -49,4 +58,81 @@ def extract_gendf_pkza(gendf_path):
     M = 1 if 'm' in A.lower() else 0
     A = int(A.lower().split(' ')[0].split('m')[0])
     pKZA = (Z * 1000 + A) * 10 + M
-    return pKZA
\ No newline at end of file
+    return pKZA
+
+def extract_cross_sections(file, MT):
+    """
+    Parse through the contents of a GENDF file section to extract the
+        cross-section data for a specific reaction type (MT).
+    
+    Arguments:
+        file (ENDFtk.tree.File): ENDFtk file object containing a specific
+            material's cross-section data.
+        MT (int): Numerical identifier for the reaction type corresponding to
+            the file's sectional organization.
+    
+    Returns:
+        sigma_list (list): All of the cross-sections for a given reaction type
+            and material, listed as floating point numbers. If the run fails,
+            the function will just return an empty list.
+    """
+
+    section = file.section(MT).content
+
+    # Only every 2nd line starting at the 3rd line has cross-section data.
+    lines = section.split('\n')[2:-2:2]
+
+    # Extract the 3rd token and convert to more conventional string
+    # representation of a float
+    sigma_list = [
+        float(line.split(' ')[2].replace('+','E+').replace('-','E-'))
+        for line in lines
+    ]
+
+    return sigma_list
+
+def iterate_MTs(MTs, file_obj, mt_dict, pKZA):
+    """
+    Iterate through all of the MTs present in a given GENDF file to extract
+        the necessary data to be able to run ALARA.
+    
+    Arguments:
+        MTs (list of int): List of reaction types present in the GENDF file.
+        file_obj (ENDFtk.tree.File): ENDFtk file object containing the
+            contents for a specific material's cross-section data.
+        mt_dict (dict): Dictionary formatted data structure for mt_table.csv
+        pKZA (int): Parent KZA identifier.
+
+    Returns:
+        gendf_data (pandas.core.frame.DataFrame): Pandas DataFrame containing
+            parent KZA values, daughter KZA values, emitted particles,
+            counts of the number of non-zero groups in the Vitamin-J groupwise
+            structure, and the cross-section values for those groups.
+    """
+
+    cross_sections_by_MT    =    []
+    emitted_particles_list  =    []
+    dKZAs                   =    []
+    groups                  =    []
+
+    for MT in MTs:
+        sigma_list = extract_cross_sections(file_obj, MT)
+        if mt_dict[MT]['delKZA'] == 'N/A':
+            dKZA = 'N/A'
+        else:
+            dKZA = pKZA - mt_dict[MT]['delKZA']
+        emitted_particles = mt_dict[MT]['Emitted Particles']
+        cross_sections_by_MT.append(sigma_list)
+        dKZAs.append(dKZA)
+        emitted_particles_list.append(emitted_particles)
+        groups.append(len(sigma_list))
+
+    gendf_data = pd.DataFrame({
```

Comment:
curious to see how this data frame looks.  I thought you needed to have a list of dictionaries to initialize it this way, or use `from_dict()`:
![image](https://github.com/user-attachments/assets/bda0829f-168f-46aa-a166-6797a521f049)

Reply:
Although I am now questioning whether the calculated daughters match the emitted particle lists - back to the tests...

Reply:
Or maybe this is a sign that it's not working??? I don't know why an emitted particle of 2n in row 4 results in a nuclide that is one nucleon heavier?  And rows 11-20 indicate an emission of n but have a nuclide one lighter (and various isomers)

Reply:
You're right and I hadn't noticed this before. I'm guessing if it is calculating incorrectly, our issue is probably coming from how we are tracking the particles, so I'll go back and look through that function to see if there is something that we missed there.

Reply:
The tests for that seem to be capturing that correctly, though...  You may want to debug by looking at the data before and after you put it into the pandas dataframe and/or read the CSV

Reply:
Ok, I'll check that out.

Reply:
So far it seems that the daughter KZAs are not changed at all from how they are calculated in the for-loop to how they appear in the Data Frame, but I'll keep digging.

Reply:
I think I got it! In line `118` where I calculate the dKZA, it is currently written as `dKZA = pKZA - mt_dict[MT]['delKZA']`, but the way that we included the delKZA values in the dictionary, they are all negative, so really it should be `dKZA = pKZA + mt_dict[MT]['delKZA']`, which seems to answer the question of why we are seeing impossible nucleon math.